### PR TITLE
Improve TLS v1.3 time rollover support and fixes for NO_ASN_TIME

### PR DESCRIPTION
**Pull request by dgarske**
* Resolves issue with millisecond time overflowing in 49 days. ZD 11848
* Fixes for building with `NO_ASN_TIME`. If used with TLS user must supply `LowResTimer` and `TimeNowInMilliseconds`.

### CODE REVIEW

**Comment by dgarske** [comment]
I’m going to rethink using time_t. I believe this can be better resolved by handling the rollover cases better. This code is only used for TLS v1.3 session tickets.

**Comment by rayozzie** [comment]
By the way - in case you ever have the need to do so - the magic way in IAR of using a 64-bit time_t is _DLIB_TIME_USES_64=1 in the project's preprocessor vars.

**Comment by dgarske** [comment]
Hi @rayozzie , I've improved this PR to better deal with rollover on the TLS v1.3. Looking forward to your feedback. David

**Comment by rayozzie** [comment]
This looks good, David!

File: src/tls13.c [hunk 1]
```diff
@@ -3540,7 +3547,10 @@ static int DoPreSharedKeys(WOLFSSL* ssl, const byte* input, word32 helloSz,
             now = TimeNowInMilliseconds();
             if (now == (word32)GETTIME_ERROR)
                 return now;
-            diff = now - ssl->session.ticketSeen;
+            if (now < ssl->session.ticketSeen)
+                diff = (0xFFFFFFFFU - ssl->session.ticketSeen) + now;
```
Comment (SparkiDev) [inline src/tls13.c:3551]: Why is this needed?
This will put it off by 1 millisecond.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/tls13.c (hunk 1 above)
Comment (dgarske) [reply src/tls13.c]: If the time in milliseconds elapses 49 days then it will possibly rollover. This will make sure the new value is calculated correctly. I adjusted the off by 1 math error. Thanks for spotting that.

Comment (SparkiDev) [reply src/tls13.c]: Rollover in TLS 1.3 should be fine.
diff has some values added and subtracted to return it to 0 base.
The only time I can see this change making a difference is when 'int' is 64 bits.

Comment (dgarske) [reply src/tls13.c]: @SparkiDev , to me it looks like `ssl->session.ticketSeen` is set with a value from `TimeNowInMilliseconds()`, thus would need to handle the 49 day rollover case?

**Review by SparkiDev — APPROVED** [review]

**Review by SparkiDev — APPROVED** [review]
